src/bot/commands/user_commands.py

```diff
@@ -33,26 +34,53 @@ def run_register_command(update, context):
 
 def run_get_player_recents_command(update, context):
     chat_id = update.message.chat_id
-    telegram_handle = update.message.from_user.username
 
-    registered_user = user_services.lookup_user_by_telegram_handle(telegram_handle)
+    # Assume default if no arguments are given
+    if not context.args:
+        registered_user = user_services.lookup_user_by_telegram_handle(
+            update.message.from_user.username
+        )
+    else:
+        # Try to match the first argument against the database for usernames
+        # If the result is not a valid user, try the second one
+        # If the result still isn't valid, assume the user's own name
+        try:
+            registered_user = user_services.lookup_user_by_telegram_handle(
```

Comment:
lookup_user doesn't error when it fails to find a user. wrapping this in a try catch doesn't do anything. user lookup will return None if nothing is found